### PR TITLE
Create Photo Model

### PR DESCRIPTION
**Pull request by mdkrog**
Create Photo Model
---
* Photos contain two fields
  * Url (string and required)
  * Caption (string and not required)
* Add a belongs to user relation
* Add foreign key to keep database integrity
* Add a has_many photos relation to User

User can add Photo
---
* Add a controller for the photo new, create and show actions
* Photo show page displays the picture and the caption
* Photo new page, has a form that allows a user to save a photo url
and caption.
* Photo must also have an associated user, defined as the current user in
the controller action.
* Add devise request helper for making it easy to login a user
during our tests.
* Test that a guest cannot add a new photo
* Update the user factory to contain basic information (email/password)

### CODE REVIEW

File: spec/models/photo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Photo, type: :model do
+  it { is_expected.to respond_to(:url) }
+  it { is_expected.to respond_to(:caption) }
+
+  it { is_expected.to validate_presence_of(:url) }
+
+  it { is_expected.to belong_to(:user) }
+end
```
Comment (mdkrog) [inline spec/models/photo_spec.rb]: I'm not sure why this is still here? I have corrected this.

Comment (antwonlee) [inline spec/models/photo_spec.rb]: Sometimes, I just ignore houndci.  =)  Great job @mdkrog.  Looks great.

Comment (thedanotto) [inline spec/models/photo_spec.rb]: Can someone merge this in? I am currently not smart enough.

Also I noticed you adjusted the version on Devise, which I'm not opposed to. This kinda brings up a bigger question in my mind, when should something be a separate pull request? IE, should the upgrade of Devise be in a different pull request? Why or why not?

Comment (mdkrog) [inline spec/models/photo_spec.rb]: Okay great, I'll merge this in. Brakeman was complaining (circleci) was failing and required Devise to be upgraded, so felt like I had to put it in this PR. I have set them as separate commits, so when we merge into master, they will be separate.
